### PR TITLE
[WIP] Fix DeepseekV3ModelTest::test_torch_compile_for_training

### PR DESCRIPTION
**Pull request by ivarflakstad**
`DeepseekV3ModelTest::test_torch_compile_for_training` fails with `torch._dynamo.exc.Unsupported: Dynamic shape operator`. Attempting to remedy by  `torch._dynamo.config` flags.

Pytorch was struggling to capture the graph when we were using `if token_indices.numel() > 0:`, so I switched to only loop over the experts that are matching from the expert mask. Should be an improvement regardless of capturing.

### CODE REVIEW

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

**Comment by ivarflakstad** [comment]
run-slow: deepseek_v3

File: src/transformers/models/deepseek_v3/modular_deepseek_v3.py [hunk 1]
```diff
@@ -173,20 +173,21 @@ def moe(self, hidden_states: torch.Tensor, topk_indices: torch.Tensor, topk_weig
         expert_mask = torch.nn.functional.one_hot(topk_indices, num_classes=len(self.experts))
         expert_mask = expert_mask.permute(2, 0, 1)
 
-        for expert_idx in range(len(self.experts)):
+        # Loop over all available experts in the model and perform the computation on each expert
```
Comment (ArthurZucker) [inline src/transformers/models/deepseek_v3/modular_deepseek_v3.py:176]: available or matched experts?

**Review by ArthurZucker — COMMENTED** [review]
You need to run `python utils/modular_model_converter.py --files deepseek3` to change the modeling otherwise test will never be fit
